### PR TITLE
lxdmetadata: connect a substitution database 

### PR DESCRIPTION
**Pull request by gabrielmougard**
This fix enables the `lxd-metadata` tool to be able to do simple template rendering operations.

Any `{{(.*?)}}` in a longdesc field will now be rendered with its substitution value, if the inner substitution key is found. Else, this template will not be rendered.

### CODE REVIEW

**Review by tomponline — APPROVED** [review]

**Review by tomponline — CHANGES_REQUESTED** [review]
Please can you fix those static analysis errors, it looks like you're forcefully killing the program rather than returning an error up to main and logging it.

**Review by ru-fu — DISMISSED** [review]
Looks good to me, just a suggestion.

File: lxd/lxd-metadata/lxd_metadata.go [hunk 1]
```diff
@@ -17,12 +18,15 @@ import (
...
 	"github.com/canonical/lxd/shared"
+
+	"gopkg.in/yaml.v2"
 )
 
 var (
 	globalLxdDocRegex   = regexp.MustCompile(`(?m)lxdmeta:generate\((.*)\)([\S\s]+)\s+---\n([\S\s]+)`)
 	lxdDocMetadataRegex = regexp.MustCompile(`(?m)([^;\s]+)=([^;\s]+)`)
 	lxdDocDataRegex     = regexp.MustCompile(`(?m)([\S]+):[\s]+([\S \"\']+)`)
+	tpl                 = regexp.MustCompile(`{{(.*?)}}`)
```
[the remark this answers is not in the document]
Comment (gabrielmougard) [reply lxd/lxd-metadata/lxd_metadata.go:29]: Could you define 'realistic values' ?

[the remark this answers is not in the document]
Comment (ru-fu) [reply lxd/lxd-metadata/lxd_metadata.go:28]: As I suggested above - not empty, and not starting with a space.

[the remark this answers is not in the document]
Comment (gabrielmougard) [reply lxd/lxd-metadata/lxd_metadata.go]: Updated. This new regex should avoid `{{}}`, `{{blah}}`, `{{ blah}}`, `{{blah }}`

[the remark this answers is not in the document]
Comment (ru-fu) [reply lxd/lxd-metadata/lxd_metadata.go]: But `{{blah}}` is what we want? Why is it still working then?

[the remark this answers is not in the document]
Comment (ru-fu) [reply lxd/lxd-metadata/lxd_metadata.go]: Substitutions will NOT have any surrounding spaces.

[the remark this answers is not in the document]
Comment (gabrielmougard) [reply lxd/lxd-metadata/lxd_metadata.go]: Sorry for the misunderstanding. Now it should avoid the surrounding spaces.

[the remark this answers is not in the document]
Comment (ru-fu) [reply lxd/lxd-metadata/lxd_metadata.go]: Now it doesn't allow any spaces at all in the string, or am I reading the expression wrong?
This is maybe a bit stricter than it needs to be - but it's unlikely we'll ever use something like ``{{two words}}`` (but use ``{{two_words}}`` instead), so it should be okay.

**Review by ru-fu — APPROVED** [review]
Thanks!

**Review by tomponline — APPROVED** [review]

**Pull request merged**